File: docs/components_page/components/offcanvas/backdrop.jl
using DashBootstrapComponents, DashHtmlComponents

backdrop_selector = html_div(
    [
        dbc_label("Backdrop:"),
        dbc_radioitems(
            id = "offcanvas-backdrop-selector",
            options = [
                Dict("label" => "True (default)", "value" => true),
                Dict("label" => "False", "value" => false),
                Dict("label" => "Static (no dismiss)", "value" => "static"),
            ],
            inline = true,
            value = true,
        ),
    ],
    className = "mb-2",
)

offcanvas = html_div([
    backdrop_selector,
    dbc_button("Open backdrop offcanvas", id = "open-offcanvas-backdrop", n_clicks = 0),
    dbc_offcanvas(
        html_p("Here is some content..."),
        id = "offcanvas-backdrop",
        title = "Offcanvas with/without backdrop",
        is_open = false,
    ),
])


callback!(
    app,
    Output("offcanvas-backdrop", "backdrop"),
    Input("offcanvas-backdrop-selector", "value"),
) do backdrop
    return backdrop
end;


callback!(
    app,
    Output("offcanvas-backdrop", "is_open"),
    Input("open-offcanvas-backdrop", "n_clicks"),
    State("offcanvas-backdrop", "is_open"),
) do n1, is_open
    return n1 > 0 ? is_open == 0 : is_open
end;

File: docs/components_page/components/offcanvas/backdrop.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

backdrop_selector <- htmlDiv(
  list(
    dbcLabel("Backdrop:"),
    dbcRadioItems(
      id = "offcanvas-backdrop-selector",
      options = list(
        list("label" = "True (default)", "value" = TRUE),
        list("label" = "False", "value" = FALSE),
        list("label" = "Static (no dismiss)", "value" = "static")
      ),
      inline = TRUE,
      value = TRUE
    )
  ),
  className = "mb-2"
)


offcanvas <- htmlDiv(
  list(
    backdrop_selector,
    dbcButton(
      "Open backdrop offcanvas",
      id = "open-offcanvas-backdrop", n_clicks = 0
    ),
    dbcOffcanvas(
      htmlP("Here is some content..."),
      id = "offcanvas-backdrop",
      title = "Offcanvas with/without backdrop",
      is_open = FALSE
    )
  )
)


app$callback(
  output("offcanvas-backdrop", "backdrop"),
  list(input("offcanvas-backdrop-selector", "value")),
  function(backdrop) {
    return(backdrop)
  }
)


app$callback(
  output("offcanvas-backdrop", "is_open"),
  list(
    input("open-offcanvas-backdrop", "n_clicks"),
    state("offcanvas-backdrop", "is_open")
  ),
  function(n1, is_open) {
    if (n1 > 0) {
      return(!is_open)
    }
    return(is_open)
  }
)
